### PR TITLE
Teleport and performance optimizations

### PR DESCRIPTION
**Pull request by hermensbas**
The general idea is to supress the aggressive randomTeleport timings without losing functionality or bot behaviour. And thereby lowering the server resources even more so. Still testing, maxing out around 7000 bots atm due hitting my RAM limits. 

Using 4000 bots as baseline for testing:

image2
image

### CODE REVIEW

**Comment by hermensbas** [comment]
image

Just happy <:

**Comment by hermensbas** [comment]
image

**Comment by EricksOliveira** [comment]
Just a doubt. What is your machine configuration?

**Comment by hermensbas** [comment]
> Just a doubt. What is your machine configuration?

An heavily undervolt 5700x (max 70 watts) 32 GB, vmware unbuntu server 24.04 minimaal, mysql 8.4.2, assigned 8 cores, 20GB to the VM.

[Config](https://github.com/hermensbas/azerothcore_installer/tree/main/config)

**Comment by hermensbas** [comment]
> image

These numbers are coming from Marco_92, 7900x, 64GB mem. But on windows, which kinda soso performance wise. He is recompiling now with the second part of the fix, no idea what numbers he will be able to push out. For me (5700x, 32GB) its around 6000 bots with decent numbers.

**Comment by hermensbas** [comment]
image
image

**Comment by hermensbas** [comment]
Marco_92, amd 7900x, 64GB, Windows
image

**Comment by hermensbas** [comment]
Marco_92 @ 12k bots
image

**Comment by hermensbas** [comment]
Seems to good to be true, i've looked at bots how active they are and whatnot, cant find any issue's though. No laggs or delays.

**Comment by hermensbas** [comment]
@Marco_92 has been testing the limits with these last couple of perf changes, laggy barely playable but playable. Obviously making the world feel alive is not equal the amount of bots in the world but mainly the behaviour of bots make the world feel alive. Even with 20k bots stormwind isnt as full as one would expect or even how busy it is with 5k bots with cmangos bots. But it doesnt hurt to optimize the process around it. The rest will hopefully follow overtime.
image

**Comment by liyunfan1223** [comment]
As far as I know, there appears to be an issue with BotActiveAlone implementation. If this value is not set to 100, it possibly causes bots to only act when there are players within 300 yards nearby. This means the bot cannot grind effectively to level up. While this functionality is useful, it doesn't match the intended purpose of this configuration option. 

By the way, the issue should be in PlayerbotAI::AllowActive(). The diff used the maximum value since server startup instead of the maximum value within a time window. This could lead to the server being judged as always under high load.

https://github.com/liyunfan1223/mod-playerbots/blob/69c3568fb66959adaf671143fe5fbded6b12a859/src/PlayerbotAI.cpp#L4018

File: conf/playerbots.conf.dist [hunk 1]
```diff
@@ -1460,7 +1460,7 @@ AiPlayerbot.RandombotsWalkingRPG.InDoors = 0
 
 # Specify percent of active bots
 # The default is 10. With 10% of all bots going active or inactive each minute.
-AiPlayerbot.BotActiveAlone = 100
+AiPlayerbot.BotActiveAlone = 90
```
Comment (liyunfan1223) [inline conf/playerbots.conf.dist]: Maybe we can keep this value default at 100 for now, and merge other optimization of teleport first.

Comment (hermensbas) [reply conf/playerbots.conf.dist]: I was kinda considering that as well, i will change it :)

Comment (hermensbas) [reply conf/playerbots.conf.dist]: ~~I did include a sync correction of the orginal value of 100 in 'src/PlayerbotAIConfig.cpp', the conf said 100 and code said 10. Or do you prefer that change in a seperate PR aswell?~~

Lets keep it clean, made another PR for the above :) So both values are set on 100.

**Comment by hermensbas** [comment]
> As far as I know, there appears to be an issue with BotActiveAlone implementation. If this value is not set to 100, it possibly causes bots to only act when there are players within 300 yards nearby. This means the bot cannot grind effectively to level up. While this functionality is useful, it doesn't match the intended purpose of this configuration option.
> 
> By the way, the issue should be in PlayerbotAI::AllowActive(). The diff used the maximum value since server startup instead of the maximum value within a time window. This could lead to the server being judged as always under high load.
> 
> https://github.com/liyunfan1223/mod-playerbots/blob/69c3568fb66959adaf671143fe5fbded6b12a859/src/PlayerbotAI.cpp#L4018

Just noticed this msg, no idea how missed your wholw feedback reply. I will look into the allowActive, i did notice the function and me doubt where it should be added. But totally makes sense the intend fits alot better then in a method called processbot

**Pull request merged**